### ISSUE TITLE
Amend description of File selector in Command configuration panel

### ISSUE THREAD

**Issue by clecap** [texstudio-org/texstudio#1148]
### Environment
- TeXstudio: 3.00.00

### Expected behavior
User manual describes the "File selector" features which show up in the advanced options of "Preferences->Command" under "Special chars".

### Actual behavior
User manual currently does not even mention the features. The feature is only described (way too short) in the UI and then only after you click "Show Advanced Options". A better description, and particularly some examples using the ext, a, m, e, r and * patterns would be helpful.

Currently problems with more advanced path settings produce Stack Exchange questions such as:  [stackexchange](https://tex.stackexchange.com/questions/471941/how-to-assign-a-new-folder-path-for-makeindex-in-texstudio)

Particularly unclear seem the * and ?* mechanism. It looks like they do not do what most readers expect (see above stackexchange entry); in particular they seem different from the more familiar regexp mechanism in shells. After 1 hour of researching I gave up trying to find a solution for a similar problem as mentioned in stack exchange.

**Comment by muzimuzhi** [comment]
> User manual describes the "File selector" features which show up in the advanced options of "Preferences->Command" under "Special chars".

It is documented in [user manual, section 4.4](https://htmlpreview.github.io/?https://github.com/texstudio-org/texstudio/blob/master/utilities/manual/usermanual_en.html#SECTION33a).

Following the linked TeX-SX question, I tested both `makeidx` and `imakeidx` packages, both works by running the following user commands in turn,
```
pdflatex -synctex=1 -interaction=nonstopmode -output-dir=_build %.tex
makeindex _build/%.idx
pdflatex -synctex=1 -interaction=nonstopmode -output-dir=_build %.tex
```
and setting "Configure TeXstudio -> Build -> Additional Search Paths -> PDF File" to "_build".
It seems there is no need to use more advanced path settings like `?[selector][pathname parts][terminating char]`.

Which specific problem did you encounter?

**Comment by muzimuzhi** [comment]
BTW, there were fixed problems related to TeXstudio's additional search path utility, see #855 and #856. You need to install [TeXstudio 3.0.0beta1](https://github.com/texstudio-org/texstudio/releases/tag/3.0.0beta1) to have those fixes.

**Comment by clecap** [comment]
Ah. I see. I have to reformulate the issue: **Link to the most up to date user manual in github index page.**

The manual you are linking to at [https://htmlpreview.github.io/?https://github.com/texstudio-org/texstudio/blob/master/utilities/manual/usermanual_en.html](url) describes this in section 4.4 External Commands.

I was looking at the probably rather old manual at [http://texstudio.sourceforge.net/manual/current/usermanual_en.html](url) which does not even have that content under section 4.4 :-(

The reason I was looking at that rather old manual is that this github instance itself at [https://github.com/texstudio-org/texstudio](url) is linking to this probably rather old manual, ironically in its Contribute: Update or extend the user manual :-) bullet point. **I suggest that this link is updated.**

The specific problem I encountered with the documentation was that I somehow misread the remark in the UI, where it says "?*aux: all .aux files in the current directory" and therefore, since I had several index files, I also used `makeindex .build/?%.idx` probably erroneously since the effect I saw in the shell just did not correspond to my understanding of the documentation. I still do not completely get the difference between a ?%.idx and a %.idx substitution :-(

I solved the specific problem with my TeX project by throwing out makeindex from the configuration file completely and replacing it with my own /bin/bash script which does a bit of directory traversal on its own. Since I am now also feeding a style file  into makeindex (and use 3 index files) this probably is the only solution for my case anyhow.
